lib/connection_controller.go

```diff
@@ -57,16 +77,45 @@ func NewConnectionController(params *DeSoParams, cmgr *ConnectionManager, handsh
 		handshake:                             handshakeController,
 		rnManager:                             rnManager,
 		AddrMgr:                               addrMgr,
+		connectIps:                            connectIps,
+		persistentIpToRemoteNodeIdsMap:        make(map[string]RemoteNodeId),
 		targetNonValidatorOutboundRemoteNodes: targetNonValidatorOutboundRemoteNodes,
 		targetNonValidatorInboundRemoteNodes:  targetNonValidatorInboundRemoteNodes,
 		limitOneInboundRemoteNodePerIP:        limitOneInboundConnectionPerIP,
+		exitChan:                              make(chan struct{}),
 	}
 }
 
+func (cc *ConnectionController) Start() {
+	cc.startGroup.Add(1)
+	go cc.startPersistentConnector()
+
+	cc.startGroup.Wait()
+	cc.exitGroup.Add(1)
+}
+
+func (cc *ConnectionController) Stop() {
+	close(cc.exitChan)
+	cc.exitGroup.Wait()
+}
+
 func (cc *ConnectionController) GetRemoteNodeManager() *RemoteNodeManager {
 	return cc.rnManager
 }
 
+func (cc *ConnectionController) startPersistentConnector() {
+	cc.startGroup.Done()
+	for {
+		select {
+		case <-cc.exitChan:
+			cc.exitGroup.Done()
+			return
+		case <-time.After(1 * time.Second):
```

Comment:
What was the connectIps refresh timer in the old code? It'll be good to set this cadence as a flag if possible too

Reply:
I'll update it in #945.